### PR TITLE
Add provision to use two different functions with same name and different arguments.

### PR DESCRIPTION
**Pull request by Pranavchiku**
Fixes https://github.com/lfortran/lfortran/issues/1266.

### CODE REVIEW

File: src/libasr/codegen/asr_to_llvm.cpp [hunk 1]
```diff
@@ -3554,7 +3554,27 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
...
+                            if ( arg->m_v == item.second ) {
+                                interface_as_arg = true;
+                                llvm::Function::arg_iterator arg_it = F->arg_begin();
+                                for (size_t j=0; j<i; j++) {
+                                    arg_it++;
+                                }
+                                llvm::FunctionType* fntype = get_function_type(*v);
+                                llvm::Function* fn = llvm::Function::Create(fntype, llvm::Function::ExternalLinkage, v->m_name, module.get());
+                                uint32_t hash = get_hash((ASR::asr_t*)v);
+                                llvm_symtab_fn[hash] = fn;
```
Comment (Pranavchiku) [inline src/libasr/codegen/asr_to_llvm.cpp:3571]: I am not sure whether it is correct solution or not, I tried to fetch the cases where nested function is passed as an argument, and then as per what Gagandeep informed about LLVM automatically manages conflict, I simply bypassed the check if "llvm_symtab_fn_names" contains it or not, as I am dealing with interfaces here so creating separate "declaration" for each interface.

In short, with this PR, I am creating unique LLVM declaration for each Interface passed as an argument.

**Comment by certik** [comment]
@czgdp1807 can you please review this one? We need the code to work, but we don't know if this is the right fix.

File: src/libasr/codegen/asr_to_llvm.cpp [hunk 2]
```diff
@@ -3554,7 +3554,27 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
...
+                    bool interface_as_arg = false;
+                    for (size_t i=0; i<x.n_args; i++) {
+                        if (is_a<ASR::Var_t>(*x.m_args[i])) {
+                            ASR::Var_t *arg = down_cast<ASR::Var_t>(x.m_args[i]);
+                            if ( arg->m_v == item.second ) {
+                                interface_as_arg = true;
+                                llvm::Function::arg_iterator arg_it = F->arg_begin();
+                                for (size_t j=0; j<i; j++) {
+                                    arg_it++;
+                                }
```
Comment (czgdp1807) [inline src/libasr/codegen/asr_to_llvm.cpp:3567]: What is the use of this block? Where is `arg_it` used?

Comment (Pranavchiku) [reply src/libasr/codegen/asr_to_llvm.cpp:3567]: Ahh I forgot to remove it, I used it previously.

**Review by czgdp1807 — APPROVED** [review]

**Pull request merged**